src/pyaps3/ecmwfml2pl.py

```
@@ -0,0 +1,259 @@
+    9617.515625, 8880.453125, 8163.375, 7470.34375, 6804.421875, 6168.53125, 5564.382813, 4993.796875, 4457.375,
+    3955.960938, 3489.234375, 3057.265625, 2659.140625, 2294.242188, 1961.5, 1659.476563, 1387.546875, 1143.25,
+    926.507813, 734.992188, 568.0625, 424.414063, 302.476563, 202.484375, 122.101563, 62.78125, 22.835938,
+    3.757813, 0.0, 0.0])
+
+PV_COEFF_B = np.array([
+    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+    0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 7e-06, 2.4e-05, 5.9e-05,
+    0.000112, 0.000199, 0.00034, 0.000562, 0.00089, 0.001353, 0.001992, 0.002857, 0.003971, 0.005378, 0.007133,
+    0.009261, 0.011806, 0.014816, 0.018318, 0.022355, 0.026964, 0.032176, 0.038026, 0.044548, 0.051773, 0.059728,
+    0.068448, 0.077958, 0.088286, 0.099462, 0.111505, 0.124448, 0.138313, 0.153125, 0.16891, 0.185689, 0.203491,
+    0.222333, 0.242244, 0.263242, 0.285354, 0.308598, 0.332939, 0.358254, 0.384363, 0.411125, 0.438391, 0.466003,
+    0.4938, 0.521619, 0.549301, 0.576692, 0.603648, 0.630036, 0.655736, 0.680643, 0.704669, 0.727739, 0.749797,
+    0.770798, 0.790717, 0.809536, 0.827256, 0.843881, 0.859432, 0.873929, 0.887408, 0.8999, 0.911448, 0.922096,
+    0.931881, 0.94086, 0.949064, 0.95655, 0.963352, 0.969513, 0.975078, 0.980072, 0.984542, 0.9885, 0.991984, 0.995003,
+    0.99763, 1.0])
+
+def get_input_variable_list(fin):
+    ds = xr.open_dataset(fin)
+    var_list = list(ds.keys())
+    ds = None
+    var_list_unique = list(set(var_list))
+    if 'lnsp' not in var_list_unique:
+      print("Error - lnsp variable missing from input file -exiting")
+      sys.exit()
+    if len(var_list_unique) < 2:
+      print("Error - Data variable missing from input file -exiting")
+      sys.exit()
+    return var_list_unique
```

A couple general comments which carry throughout the module-

- For these general functions, let's raise a meaningful exception rather than call `sys.exit`
- you appear to sometimes use 2 spaces for indentation rather than 4
- a docstring describing the inputs/outputs would be helpful. for this function, it's hard to guess just from looking at these 10 lines what `fin` should be, or where i could find a file with `'lnsp'` as an `xarray`-readable variable
- As we mentioned offline, let's double check- is xarray needed for this? or is it just used to read in the inputs (which are GRIB files readable by `pygrib`?) and write the outputs

For this specific function
- this function just open a file and gets the variables, but then we reopen and use the file right after
- it we're doing this anyway, we should use the context manager version of `with xr.open_dataset(fin):...` . It's only the Python-gdal bindings that require you to do `ds = None` to close a dataset
- I can't tell what `var_list_unique` is supposed to contain that is a problem if the length is < 2

---

Thank you @scottstanie, I think we might need the netcdf reader here because the NISAR PCM has already the interface to generate the netcdf to feed the ISCE3, let me find a cheaper way to do this, maybe only the netcdf reader instead of the xarray.